src/components/date-picker/DatePicker.stories.tsx

```diff
@@ -53,6 +56,136 @@ export const manuallyEditable = () => (
     </IntlProvider>
 );
 
+export const withLimitedDateRange = () => {
+    const maxDate = new Date('February 25, 2021');
+    const sixDays = 1000 * 60 * 60 * 24 * 6;
+    const minDate = new Date(maxDate.valueOf() - sixDays);
+    const componentStore = new Store({
+        date: maxDate,
+        fromDate: null,
+        toDate: null,
+    });
+    return (
+        <State store={componentStore}>
+            {state => (
+                <IntlProvider locale="en-US">
+                    <DatePicker
+                        isTextInputAllowed
+                        placeholder="Date"
+                        label="Date Picker"
+                        minDate={minDate}
+                        maxDate={maxDate}
+                        value={state.date}
+                    />
+                </IntlProvider>
+            )}
+        </State>
+    );
+};
+
+export const alwaysVisibleWithHiddenInput = () => {
```

Comment:
Is the component keyboard-accessible in this configuration? It should be OK with the separate input, but not sure about hiding the text box with no replacement.

Reply:
Good call, @alexkrolick. I replaced the input hiding logic with a new custom input field prop and added a visual regression test that demonstrates keyboard selection. Could you review again?